wallet-new/src/Cardano/Wallet/Kernel/DB/Spec.hs

```diff
@@ -79,11 +77,23 @@ removePending ids (Pending (InDb old)) = Pending (InDb $ old `withoutKeys` ids)
 
 -- | Per-wallet state
 --
--- This is the same across all wallet types.
+-- NOTE: At the moment this does not included the expected UTxO. The expected
+-- UTxO is used for two things:
+--
+-- * Block resolution (translating tx inputs to their corresponding outputs, so
+--   that we know the corresponding addresses, needed for prefilering)
+-- * Minimum balance computation
+--
+-- Fortunately however we can rely on a full node as backing, so we don't need
+-- to use the expected UTxO for block resolution (this is explained in the
+-- formal spec in section "Prefiltering -- Consequences", under "possible
+-- alternatives"), and minimum balance computation is a new feature that we
+-- haven't implemented yet.
+--
+-- NOTE: This is the same across all wallet types.
 data Checkpoint = Checkpoint {
       _checkpointUtxo        :: InDb Core.Utxo
     , _checkpointUtxoBalance :: InDb Core.Coin
-    , _checkpointExpected    :: InDb Core.Utxo
```

Comment:
Are we tracking in a ticket anywhere that we need to tackle expected balance separately from this PR?

Reply:
Opened https://iohk.myjetbrains.com/youtrack/issue/CBR-346 .